presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``20 GB``
+ * **Description:** Serves as default value for ``query_max_memory`` session property. This property also describes strict limit of total memory allocated around the cluster that may be used to process single query. The query is dropped if the limit is reached unless session want to prevent that by setting session property ``resource_overcommit``. The session may also want to decrease system pressure, so it's possible to decrease query memory limit for session by setting ``query_max_memory`` to smaller value. Setting ``query_max_memory`` to higher value then ``query.max-memory`` will not have any effect. This property may be used to ensure that single query cannot use all resources in cluster. The value should be set to be higher than what typical expected query in system will need - that way system will be resistant to SQL bugs that would cause large unwanted computation. Also if rare queries will require more memory, then the ``resource_overcommit`` session property may be used to break the limit. It is important to set this value to higher then default when presto runs complex queries on large datasets.
+
+
+``query.max-memory-per-node``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``JVM max memory`` * ``0.1``
+ * **Description:** The purpose of that is same as of :ref:`query.max-memory<query-max-memory>` but the memory is not counted cluster-wise but node-wise instead.
+
+
+``query.max-queued-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``5000``
+ * **Description:** **Deprecated** Describes how many queries may wait in worker queue. If the limit is reached master server will consider worker blocked and will not push more tasks to him. Setting this value high may allow to order a lot of queries at once with the cost of additional memory needed to keep informations about tasks to process. Lowering this value will decrease system capacity but will allow to utilize memore for real processing of date instead of queuing. It shouldn't be used in new configuration, the ``query.queue-config-file`` can be used instead.
+
+
+``query.max-run-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``100 d``
+ * **Description:** Used as default for session property ``query_max_run_time``. If the presto works in environment where there are mostly very long queries (over 100 days) than it may be a good idea to increase this value to avoid dropping clients that didn't set their session property correctly. On the other hand in the presto works in environment where they are only very short queries this value set to small value may be used to detect user errors in queries. It may also be decreased in poor presto cluster configuration with mostly short queries to increase garbage collection efficiency and by that lowering memory usage in cluster.
```

Comment:
There's some cleaned up language below, but some broader questions: Is it expected that people will have queries that run for more than 100 days or anywhere close to that?  And you'd have to decrease it substantially in order for it to be of any real use in catching user errors.  Also if you have a bad cluster configuration seems like you'd be better off giving more memory to Presto (so that it is configured properly for the workload) then whatever gains you would get with this.

Used as the default for the ``query_max_run_time`` session property.  If Presto is mostly used to process very long running queries (over 100 days), then it may be a good idea to increase this value.  On the other hand, if there are only very short queries then decreasing this value may help detect user errors in queries.  It may also be helpful in a poorly configured Presto cluster to increase garbage collection efficiency and thereby lower memory usage in the cluster.